lib/rubocop/cop/layout/space_inside_array_literal_brackets.rb

```diff
@@ -44,8 +44,8 @@ class SpaceInsideArrayLiteralBrackets < Cop
         EMPTY_MSG = '%<command>s space inside empty array brackets.'.freeze
 
         def on_array(node)
+          return unless node.square_brackets?
           left, right = array_brackets(node)
-          return if right.nil?
```

Comment:
Instead of checking `nil?` on the right bracket, I've swapped it with an existing safety check. This is an array_node method used similarly in the `Style/TrailingCommaInLiteral` cop [right here](https://github.com/bbatsov/rubocop/blob/master/lib/rubocop/cop/style/trailing_comma_in_literal.rb#L48).

Since this cop is specific to Arrays defined with brackets, this check is more logical. The guard clause successfully ignores arrays defined with `::new`, e.g. `Array.new(4)`.  This is why there were 3 `cli_autocorrect_spec` failures... My first hypothesis was wrong.